src/app/tasks/create-task/create-task.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, EventEmitter, Output } from '@angular/core';
+import { Task, emptyTask } from '../task.model';
+import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { TaskService } from '../task.service';
+import { Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-create-task',
+  standalone: true,
+  imports: [
+    ReactiveFormsModule
+  ],
+  templateUrl: './create-task.component.html',
+  styleUrl: './create-task.component.scss'
+})
+export class CreateTaskComponent {
+  
+  @Output() taskCreatedEvent = new EventEmitter<Task>();
+
+  createTaskForm: FormGroup = new FormGroup({
+    'title': new FormControl(null, [
+      Validators.required,
+      this.validateTitleUnique.bind(this)
+    ]), //Custom validator for unique title
+    'description': new FormControl(),
+    'type': new FormControl(null, Validators.required),
+    'createdOn': new FormControl(new Date().toDateString()),
+    'status': new FormControl('In Progress', Validators.required)
+  });
+  
+  newTask: Task = emptyTask;
+
+  constructor(private taskService: TaskService) {}
+
+  onSubmit () {
+    this.newTask.title = this.createTaskForm.get('title')!.value; 
+    this.newTask.description = this.createTaskForm.get('description')!.value; 
```

Comment:
you can create a task object inside of this method instead of declaring it outside also don't use non-null assertion (!) here instead you can use ?